.github/workflows/upload-celocli-executables.yml

```diff
@@ -16,9 +16,141 @@ on:
         required: true
 
 jobs:
+  setupVariables:
+    name: Asserts and setup
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.setupVars.outputs.version }}
+      commit: ${{ steps.setupVars.outputs.commit }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Setup Node.js 20.x
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+
+      - name: 'enable corepack for yarn'
+        run: sudo corepack enable yarn
+        shell: bash
+      # must call twice because of chicken and egg problem with yarn and node
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'yarn'
+
+      - name: Install Dependencies
+        shell: bash
+        run: yarn
```

Comment:
it seems this job is to get the version in variable. 

do we really need to run install here to do that?

Reply:
maybe not